stablehlo/reference/Ops.cpp

```diff
@@ -102,6 +102,81 @@ Tensor evalSliceOp(const Tensor &operand, const Sizes &startIndices,
                      inferredTypes[0].cast<ShapedType>());
 }
 
+static Tensor computeSum(const Tensor &operand, const Tensor &initValue,
+                         const Axes &dimensions, ShapedType resultType) {
+  Tensor result(resultType);
+  for (auto resultIt = result.index_begin(); resultIt != result.index_end();
+       ++resultIt)
+    result.set(*resultIt, initValue.get({}));
+
+  for (auto operandIt = operand.index_begin(); operandIt != operand.index_end();
+       ++operandIt) {
+    Index resultIndex = *operandIt;
+    for (auto dim : llvm::reverse(dimensions))
+      resultIndex.erase(resultIndex.begin() + dim);
+
+    Tensor currVal(initValue.getType());
+    currVal.set({}, result.get(resultIndex));
+
+    Tensor nextValToAdd(initValue.getType());
+    nextValToAdd.set({}, operand.get(*operandIt));
+
+    auto addResult = evalAddOp(currVal, nextValToAdd, currVal.getType());
+    result.set(resultIndex, addResult.get({}));
+  }
+  return result;
+}
```

Comment:
In the code snippet above, I also renamed `operand` to `input` to follow `evalReduceOp` and changed the logic of computing `resultIndex` which was changed during review of the corresponding PR.